test-framework/ui/src/main/java/org/keycloak/testframework/ui/page/AbstractPage.java

```diff
@@ -38,4 +46,8 @@ public boolean isActivePage() {
         return getExpectedPageId().equals(getCurrentPageId());
     }
 
+    public void assertCurrent() {
+        String name = getClass().getSimpleName();
+        Assertions.assertTrue(isActivePage(), "Expected " + name + " but was " + driver.getTitle() + " (" + driver.getCurrentUrl() + ")");
```

Comment:
I would just use the `data-page-id` stuff in the message here as well. Either:
```suggestion
        Assertions.assertTrue(isActivePage(), "Expected " + getExpectedPageId() + " but was " + getCurrentPageId());
```

Or, perhaps even better:
```
Assertions.assertEquals(getCurrentPageId(), getExpectedPageId());
```